api/include/opentelemetry/logs/log_record.h

```diff
@@ -82,25 +77,28 @@ struct LogRecord
   core::SystemTimestamp timestamp;  // uint64 nanoseconds since Unix epoch
   trace::TraceId trace_id;          // byte sequence
   trace::SpanId span_id;            // byte sequence
-  trace::TraceFlags trace_flag;     // byte
-  Severity severity;  // Severity enum that combines severity_text and severity_number in the
-                      // LogDataModel (can separate in SDK)
+  trace::TraceFlags trace_flags;    // byte
+  Severity severity;                // Severity enum that combines severity_text and severity_number
 
   // other fields that will not be set by default
   nostd::string_view name;  // string
   nostd::string_view body;  // currently a simple string, but should be changed "Any" type
-  common::KeyValueIterable &resource;    // key/value pair list
-  common::KeyValueIterable &attributes;  // key/value pair list
+  nostd::shared_ptr<common::KeyValueIterable> resource;    // key/value pair list
+  nostd::shared_ptr<common::KeyValueIterable> attributes;  // key/value pair list
 
   /* Default log record if user does not overwrite this.
    * TODO: find better data type to represent the <Any> type for "body"
    * Future enhancement: Potentially add other constructors to take default arguments
    * from the user
    **/
-  LogRecord() : resource(_nullKV), attributes(_nullKV)
+  LogRecord()
   {
-    // TODO: in SDK, assign a default timestamp if not specified
-    name = "";
+    // Assign default values
+    timestamp   = core::SystemTimestamp(std::chrono::seconds(0));
+    severity    = Severity::kDefault;
+    trace_id    = trace::TraceId();
+    span_id     = trace::SpanId();
+    trace_flags = trace::TraceFlags();
```

Comment:
nit: these can also be initialized through initializer list.

Reply:
Just initialize these fields along with its definition line.

Reply:
Hmm could we remove the constructor as a whole? Default values are set as follows by constructors in timestamp.h, trace_id.h, span_id.h, trace_flags.h, and string_view.h respectively. I believe shared_ptrs are set to nullptr by default (tested this locally). 

So we could have: 

```
struct LogRecord
{
  core::SystemTimestamp timestamp;    // default is 0
  trace::TraceId trace_id;            // default is 00000000000000000000000000000000
  trace::SpanId span_id;              // default is 0000000000000000
  trace::TraceFlags trace_flags;      // default is 00
  Severity severity;                  // default is 0 for enum initialization
  nostd::string_view name;            // default is length of 0 and data of nullptr
  nostd::string_view body;            // default is length of 0 and data of nullptr
  nostd::shared_ptr<common::KeyValueIterable> resource;    // default is nullptr
  nostd::shared_ptr<common::KeyValueIterable> attributes;  // default is nullptr 
}
```

Also if possible, we can also initialize severity to `kInfo` as default here, instead of having it injected in the SDK.

Reply:
Yeah, that would be fine too.

Reply:
If using the default generated ctor, `Severity severity` will be initialized to 0, which is actually not an valid value in `enum class Severity`. Perhaps we could add 0 to the enum list as `kInvalid`?

Reply:
For sure, and later the SDK could inject the severity to something else like `kInfo` if it's `kInvalid`.